src/resolvers/utils.ts

```diff
@@ -0,0 +1,53 @@
+import { dirname, isAbsolute, join } from '@stoplight/path';
+import { JsonPath } from '@stoplight/types';
+
+import { extractSourceFromRef } from '../extractSourceFromRef';
+import { isPlainObject } from '../isPlainObject';
+import { assertObjectWithValidRef, assertResolvableInput, hasSomeRef } from './guards';
+
+export function applyOverrides(document: unknown, value: unknown) {
+  if (isPlainObject(value) && isPlainObject(document) && ('summary' in document || 'description' in document)) {
+    return {
+      ...value,
+      ...('description' in document ? { description: document.description } : null),
```

Comment:
I'm still curious about this function - we already had that one log where you had to override this override to retain the value's description - what are the reasons for overwriting?  Should it only be overwriting those values if value.description and value.summary is null or empty string?

Reply:
So, with that code gone, you'd have the following result

```
const doc = {
  type: 'object',
  properties: {
    greatestBear: {
      $ref: '#/$defs/Bear',
      description: 'The greatest bear!',
     },
  },
  $defs: {
    Bear: {
       type: 'string',
     },
  },
};

 expect(resolveInlineRef(doc, '#/properties/greatestBear')).toStrictEqual({
    type: 'string',
 });
 
 // instead of
 
 expect(resolveInlineRef(doc, '#/properties/greatestBear')).toStrictEqual({
   type: 'string',
   description: 'The greatest bear!',
});
```

The reason we do that override is because in JSON Schema, you can have "sibling" properties.
A so-called "sibling property" is a property next to "$ref", which, in the case above, is "description", but one could have others too.

Now, the interesting part is that how you deal with sibling properties is up to you. JSON Schema states in their specification that it's up to implementator to decide what to do with them, whether they act as an overwrite, or whether they are dropped etc.

The reason we decide to support `summary` and `description` is because of OAS 3.1 that further specifies what should be done with those "sibling properties". The OAS 3.1 spec explicitly specify which "sibling properties" to support, and that's why stuck with that.

<img width="850" alt="image" src="https://user-images.githubusercontent.com/9273484/182947146-7fb87ee6-b167-4752-8822-085bb754bcf2.png">

>  Should it only be overwriting those values if value.description and value.summary is null or empty string?

These values should generally be string or missing, so generally speaking there shouldn't be something else there.
If there is, Spectral would complain about it and a user would be told to address that issue, therefore it's less of a concern, I'd say.

Reply:
Ahh, this is my mistake, I was reading it as value being the referrer and document being the referenced component, but it's actually the opposite of that.